kv: test that verifies we don't emit duplicate intents

RaduBerinde opened the pull request:
In #5042 we had an issue with transactions that wrote a key and then wrote the
same key as part of the final batch (with commit). This change adds a test that
verifies that the TxnCoordSender does not generate duplicate intents and that it
merges intents for overlapping ranges.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5428)

<!-- Reviewable:end -->

tschottdorf commented on kv/txn_coord_sender_test.go:
```diff
@@ -807,6 +807,75 @@ func TestTxnCoordSenderReleaseTxnMeta(t *testing.T) {
+	stopper := stop.NewStopper()
+	manual := hlc.NewManualClock(0)
+	clock := hlc.NewClock(manual.UnixNano)
+	clock.SetMaxOffset(20)
```
don't need that for anything as far as I can tell.

tschottdorf commented:
LGTM. Thanks for taking care of this.

RaduBerinde committed:
kv: test that verifies we don't emit duplicate intents

In #5042 we had an issue with transactions that wrote a key and then wrote the
same key as part of the final batch (with commit). This change adds a test that
verifies that the TxnCoordSender does not generate duplicate intents and that it
merges intents for overlapping ranges.
